systemd/systemd.go

```diff
@@ -600,8 +610,29 @@ func (s *systemd) getUnitStatus(properties []string, unitNames []string) ([]*Uni
 
 	for _, bs := range statusregex.FindAllSubmatch(bs, -1) {
 		if len(bs[0]) == 0 {
+			sts = append(sts, cur)
+
+			// If we received more replies than what we requested, bail.
+			if len(sts) > len(unitNames) {
+				break
```

Comment:
this break was already a bit strange, my preference would be to move the error returning directly here from outside the loop:

```
if len(sts) => len(unitNames) {
  return nil, fmt.Errorf("cannot get unit status: got more results than expected")
}
```

and remove the if=>error after the for

one test will need changing

Reply:
Made the in loop return directly for the case where more data is returned that what we need (**>=**).

@pedronis 

However, the check outside the loop historically included the ability to detect the case where **less** data was returned that what was required, as well. If I remove that check now, that condition will not be caught. I left the second check in for the latter case, and added a unit test to verify this.

Let me know if you disagree with my interpretation.

Reply:
> However, the check outside the loop historically included the ability to detect the case where less data was returned than what was required

I missed this aspect, thanks

Reply:
Resolved. Thanks.